### PR TITLE
Add solution

### PR DESCRIPTION
**Pull request by AndrewPolishchuc**

### CODE REVIEW

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+
```
Comment (Rudiachenko) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: Extra separation

File: src/main/java/com/internet/shop/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
Comment (Rudiachenko) [inline src/main/java/com/internet/shop/model/Order.java:18]: add products to constructor

File: src/main/java/com/internet/shop/model/Product.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package com.internet.shop.model;
+
+public class Product {
+    private Long id;
+    private String name;
+    private double price;
+
+    public Product(String name, double price) {
+        this.name = name;
+        this.price = price;
```
Comment (Rudiachenko) [inline src/main/java/com/internet/shop/model/Product.java]: we should use BigDecimal when working with money.

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
Comment (Rudiachenko) [inline src/main/java/com/internet/shop/model/ShoppingCart.java:18]: add products to constructor

File: src/main/java/com/internet/shop/model/User.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.internet.shop.model;
+
+public class User {
+    private Long id;
+    private String name;
+    private String login;
+    private String password;
+
```
Comment (Rudiachenko) [inline src/main/java/com/internet/shop/model/User.java:8]: constructor, setters/getters?

**Review by Rudiachenko — COMMENTED** [review]
good job. fix it now

File: src/main/java/com/internet/shop/db/Storage.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    public static final List<ShoppingCart> shoppingCarts = new ArrayList<>();
+    public static final List<User> users = new ArrayList<>();
+    private static Long productId = 0L;
+    private static Long orderId = 0L;
+    private static Long shoppingCartId = 0L;
+    private static Long userId = 0L;
+
+    private Storage() {
+
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/db/Storage.java:23]: do you need this contructor?

File: src/main/java/com/internet/shop/model/Order.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userId;
+
+    public Order(List<Product> products, Long userId) {
+        this.products = products;
+        this.userId = userId;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/model/Order.java]: let's only initialize products list here but not pass it

File: src/main/java/com/internet/shop/model/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.internet.shop.model;
+
+import java.util.List;
+
+public class ShoppingCart {
+    private Long id;
+    private List<Product> products;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/model/ShoppingCart.java:8]: let's initialize it in constructor

**Review by dmytro-hryhoruk — COMMENTED** [review]
We should only ProductDao and ProductService in this PR

File: src/main/java/com/internet/shop/dao/UserDao.java [hunk 9]
```diff
@@ -0,0 +1,4 @@
+package com.internet.shop.dao;
+
+public interface UserDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/UserDao.java]: ?

File: src/main/java/com/internet/shop/dao/ShoppingDao.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.ShoppingCart;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface ShoppingDao {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/ShoppingDao.java]: ?

File: src/main/java/com/internet/shop/db/Storage.java (hunk 6 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/db/Storage.java:23]: ?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**